src/action/interactive.rs

```diff
@@ -171,15 +177,41 @@ impl UserPicked {
         let KeyEvent { code, modifiers } = event;
 
         match code {
-            KeyCode::Up => state.select_previous(),
-            KeyCode::Down => state.select_next(),
+            KeyCode::Left => {
+                state.cursor_index = state.cursor_index.checked_sub(1).unwrap_or_default()
+            }
+            KeyCode::Right => {
+                state.cursor_index =
+                    (state.cursor_index + 1).min(state.custom_replacement.len() as u16)
```

Comment:
Length is taken in almost all cases, add a local var above the match for clarity